### PR TITLE
Mrc 5146 container gh actions

### PR DESCRIPTION
**Pull request by M-Kusumgar**
This turned into a longer PR than I had initially hoped for. This PR builds docker and pushes it dockerhub and github container registry. Main changes include:
* workflow for pushing
* upgrading R version in the package because R versions less than 4.1.0 have problems with decoding url params
* added host option in server main entrypoint
* added worker main entrypoint that also creates their worker directory in `.packit`
* `REDIS_CONTAINER_NAME` env var so that we can connect to redis running in another container
* various scripts for testing this out fully in a docker setup, these are outlined in the README

### CODE REVIEW

File: docker/Dockerfile
Comment (M-Kusumgar) [inline docker/Dockerfile]: same issue as in outpack_server PR, gotta add this as a safe directory as it is not owned by us in the container, also the global gitignore means that the .packit where the workers are isnt tracked by git

File: docker/test/run-test
Comment (absternator) [inline docker/test/run-test:20]: Q: whats this one for?

File: docker/test/run-test
Comment (absternator) [inline docker/test/run-test:12]: also wondering if redis is needed here? or maybe another script that dosent... as this will all be brought up with packet (which already brings up redis), thus will get conflict

File: docker/Dockerfile
Comment (absternator) [inline docker/Dockerfile]: will this come from packet? or will this have another redis container?

File: docker/Dockerfile
Comment (absternator) [inline docker/Dockerfile]: a more informative name just to be safe ahha eg. `orderly.runner.queueId`

**Review by absternator — APPROVED** [review]
LGTM!!!! just some minor commnets.. 
- Maybe the scripts can be one level higher above test? in a scripts folder for consitency
- 

Also just a question... im gussiing in packet we just spin up the runner and worker right?

**Review by absternator — COMMENTED** [review]
LGTM!!!! just some minor commnets.. 
- Maybe the scripts can be one level higher above test? in a scripts folder for consitency
- 

Also just a question... im gussiing in packet we just spin up the runner and worker right?

File: docker/Dockerfile
Comment (richfitz) [inline docker/Dockerfile]: can yo sort these alphabetically please?

File: docker/Dockerfile
Comment (richfitz) [inline docker/Dockerfile]: I don't think that this should be a default envvar in the image

File: docker/test/clear-test
Comment (richfitz) [inline docker/test/clear-test]: Use `docker rm -f` and you can avoid the repetition here

File: docker/test/run-test
Comment (richfitz) [reply docker/test/run-test:12]: we do need redis here, but that's fine for testing this stuff. You will get a conflict if two containers bring up redis on the same port, but in practice it's not hard to deal with

File: docker/test/run-test
Comment (richfitz) [inline docker/test/run-test:31]:
```suggestion
       --entrypoint="/usr/local/bin/orderly.runner.server" \
       --env=REDIS_CONTAINER_NAME=orderly.runner_redis \
```

etc

File: docker/test/run-test
Comment (richfitz) [inline docker/test/run-test]:
```suggestion
set -exu
```

Everywhere you want `-e` you usually want `-u`

**Review by richfitz — CHANGES_REQUESTED** [review]

File: docker/test/run-test
Comment (M-Kusumgar) [reply docker/test/run-test:20]: that one is for a couple things:
1. you need a volume attached to a container to do anything with it so for us modifying the orderly repo, we could use this as the place to change things in the volume for testing
2. to view how things change in the working directory (since workers will be running things and producing result files and metadata of the run) theres a helper script `docker/test/copy-orderly-root` that will copy the folder so we can see it easily - that script uses this container too

File: docker/Dockerfile
Comment (M-Kusumgar) [reply docker/Dockerfile]: packit does not need a redis container, after addressing rich's comments, this will come from env args in docker run and so these variables will come from the packit deploy tool which makes sense

File: docker/Dockerfile
Comment (M-Kusumgar) [reply docker/Dockerfile]: done

File: docker/Dockerfile
Comment (M-Kusumgar) [reply docker/Dockerfile]: removed

File: docker/Dockerfile
Comment (M-Kusumgar) [reply docker/Dockerfile]: gone with `orderly.runner.queue` for casing consistency

File: docker/test/run-test
Comment (M-Kusumgar) [reply docker/test/run-test:31]: done

File: docker/test/run-test
Comment (M-Kusumgar) [reply docker/test/run-test]: done

**Comment by M-Kusumgar** [comment]
I have addressed the comments but also done a bit of cleanup, all ive done that is extra is extract out certain variables you may want to change like container names, after approval I will also change it to pull image from main and merge it in so that we arent always pulling from this random branch. I will also remove triggers to the build and push workflow from this branch.

Also I remember this ticket that I had to do involving replacing `_` with `-` in docker container names, ive forgotten why I had to do it but to be on the safe side, the naming convention is now `orderly.runner-[container]` instead of `orderly.runner_[container]`

File: R/main.R
Comment (richfitz) [inline R/main.R:35]: Does this error informatively if not set?

File: R/queue.R
Comment (richfitz) [inline R/queue.R:33]: Suggest erroring here or orderly_queue_id() here if not set?  but we could explore that in another PR if you prefer?

**Review by richfitz — APPROVED** [review]

File: R/queue.R
Comment (M-Kusumgar) [reply R/queue.R:33]: i think we should error in the `main_worker` function because this is okay if we havent set the queue id, in fact we use that behaviour a lot in the tests, have made a ticket: https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-5663/orderly.runner-worker-error-if-ORDERLYRUNNERQUEUEID-not-set

**Pull request merged**
